Unable to run demo

<!-- Enter your issue description here -->

## Details
- Browser and browser version:  Chrome 61
- OS version: 10.12.1
- xterm.js version: latest

### Steps to reproduce

1. git clone https://github.com/xtermjs/xterm.js.git
2. cd xterm.js
3. npm i
4. npm run watch
5. In another terminal "npm start"


When I do this I get:
```
~/Downloads/node_modules/xterm/typings/xterm.d.ts(14,15): error TS2300: Duplicate identifier 'FontWeight'.
~/Downloads/node_modules/xterm/typings/xterm.d.ts(247,16): error TS2300: Duplicate identifier 'Terminal'.
~/Downloads/xterm.js-master/typings/xterm.d.ts(14,15): error TS2300: Duplicate identifier 'FontWeight'.
~/Downloads/xterm.js-master/typings/xterm.d.ts(254,16): error TS2300: Duplicate identifier 'Terminal'.
```

When I open the demo I get the black terminal screen with no text.
In Chrome developer tools I see:

```
Uncaught TypeError: Cannot read property 'length' of undefined
    at Parser.parse (xterm.js:2280)
    at Terminal._innerWrite (xterm.js:4090)
    at xterm.js:4075
```

Which is coming from:
```
    Parser.prototype.parse = function (data) {
        var l = data.length;
        var cs;
        var ch;

```

--- a/src/Terminal.ts
+++ b/src/Terminal.ts
@@ -1252,6 +1252,11 @@ export class Terminal extends EventEmitter implements ITerminal, IInputHandlingT
    * @param {string} data The text to write to the terminal.
    */
   public write(data: string): void {
+    // Ignore falsy data values (including the empty string)
+    if (!data) {
+      return;
+    }
+
     this.writeBuffer.push(data);
 
     // Send XOFF to pause the pty process if the write buffer becomes too large so

